### PR TITLE
[docs][migrate] New supported options to install yb_voyager

### PR DESCRIPTION
**Pull request by aishwarya24**
resolves https://github.com/yugabyte/yb-voyager/issues/440, https://github.com/yugabyte/yugabyte-db/issues/15658, https://github.com/orgs/yugabyte/projects/10/views/11?pane=issue&itemId=17443880
Summary
- Introduced tabs for different install options
https://deploy-preview-15348--infallible-bardeen-164bc9.netlify.app/preview/migrate/install-yb-voyager/

### CODE REVIEW

File: docs/content/preview/migrate/airgapped.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+
```
Comment (polarweasel) [inline docs/content/preview/migrate/airgapped.md:1]:
```suggestion
<!--
+++
private=true
+++
-->
```

**Review by polarweasel — APPROVED** [review]
LGTM! Add the frontmatter to each included file to keep them out of the algolia/google indexes...

File: docs/content/preview/migrate/linux.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+1. Install the PostgreSQL and Oracle instant clients repositories using the following command:
+
+    ```sh
+    sudo yum install pgdg-redhat-repo oracle-instant-clients-repo
+    ```
+
+    These repositories contain the rest of the dependencies required to run `yb-voyager`.
+
+Note that if you're using **RHEL 8** or **CentOS 8**, perform the following two steps before proceeding to step 5.
```
Comment (ShivanshGahlot) [inline docs/content/preview/migrate/linux.md:43]: Can we demarcate this section a little more? Something like
`Renaming index names` or `Usage of source_db_schema` here: https://docs.yugabyte.com/preview/migrate/migrate-steps/#export-schema

File: docs/content/preview/migrate/macos.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
+<!--
++++
+private=true
++++
+-->
+
+Perform the following steps to install yb-voyager using brew for MacOS:
+
+1. [Tap](https://docs.brew.sh/Taps) to the `yugabyte` brew repository using the following command:
```
Comment (ShivanshGahlot) [inline docs/content/preview/migrate/macos.md:9]: Do we need to link the brew documentation for tap here? @amit-yb

File: docs/content/preview/migrate/airgapped.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
+<!--
++++
+private=true
++++
+-->
+
+You can perform an airgapped installation on RHEL 7/8, CentOS 7/8, and Ubuntu OS.
+
+For airgapped installations, you need to download all the rpm files that are necessary for installing and running `yb-voyager` on a machine with an internet connection.
```
Comment (ShivanshGahlot) [inline docs/content/preview/migrate/airgapped.md:9]: Please modify this line to:
`For airgapped installations, you need to download the tarball containing all the rpm files that are necessary for installing and running `yb-voyager` on a machine with an internet connection. You must download the tarball corresponding to your airgapped machine's OS. You can then transfer these files to your airgapped machine and proceed with the installation using the steps below.`

File: docs/content/preview/migrate/airgapped.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+Ensure that the OS of your current machine and the airgapped machine is the _same_ so that you can download the correct versions of the rpm files.
```
Comment (ShivanshGahlot) [inline docs/content/preview/migrate/airgapped.md:11]: You can remove this line. Its not needed anymore.

File: docs/content/preview/migrate/macos.md (hunk 3 above)
Comment (aishwarya24) [reply docs/content/preview/migrate/macos.md:9]: I included it just as a reference to a user (no assumptions that they may already know about it).

File: docs/content/preview/migrate/install-yb-voyager.md [hunk 6]
```diff
@@ -60,41 +61,68 @@ The export directory has the following sub-directories and files:
...
+    <a href="#github" class="nav-link" id="github-tab" data-toggle="tab" role="tab" aria-controls="github" aria-selected="true">
+      <i class="fab fa-github" aria-hidden="true"></i>
+      Source
+    </a>
+  </li>
+</ul>
+
+<div class="tab-content">
+  <div id="linux" class="tab-pane fade show active" role="tabpanel" aria-labelledby="linux-tab">
+{{% readFile "./linux.md" %}}
```
Comment (polarweasel) [inline docs/content/preview/migrate/install-yb-voyager.md:108]: Do this for all...
```suggestion
{{% readfile "./linux.md" %}}
```

File: docs/content/preview/migrate/install-yb-voyager.md [hunk 7]
```diff
@@ -60,41 +71,68 @@ The export directory has the following sub-directories and files:
 ## Install yb-voyager
 
 YugabyteDB Voyager consists of the yb-voyager command line executable. yb-voyager keeps all of its migration state, including exported schema and data, in a local directory called the [*export directory*](#create-an-export-directory).
-<!-- For more information, refer to [Export directory](../../yb-voyager/reference/#export-directory). -->
```
Comment (amit-yb) [inline docs/content/preview/migrate/install-yb-voyager.md:63]: This is misplaced. No need to talk about export-dir while installing voyager. I suggest to drop the entire paragraph.

File: docs/content/preview/migrate/linux.md [hunk 8]
```diff
@@ -0,0 +1,76 @@
+<!--
```
Comment (amit-yb) [inline docs/content/preview/migrate/linux.md:1]: perhaps the file can be renamed as `rhel.md`.

File: docs/content/preview/migrate/linux.md [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+    ```sh
+    sudo yum install https://s3.us-west-2.amazonaws.com/downloads.yugabyte.com/repos/reporpms/yb-yum-repo-1.0-0.noarch.rpm
+    ```
+
+    This repository contains the yb-voyager rpm and other dependencies required to run `yb-voyager`.
+
+1. Install the `epel-release` repository using the following command:
+
+    ```sh
+    // For RHEL 7 or CentOS 7
```
Comment (amit-yb) [inline docs/content/preview/migrate/linux.md:26]: Here and in all other example commands, use `#` as a comment start character.

File: docs/content/preview/migrate/install-yb-voyager.md [hunk 10]
```diff
@@ -60,41 +71,68 @@ The export directory has the following sub-directories and files:
...
+  </div>
+  <div id="docker" class="tab-pane fade" role="tabpanel" aria-labelledby="docker-tab">
+{{% readfile "./docker.md" %}}
+  </div>
+  <div id="github" class="tab-pane fade" role="tabpanel" aria-labelledby="github-tab">
+{{% readfile "./github.md" %}}
+  </div>
+</div>
 
 To learn more about yb-voyager, refer to [YugabyteDB Voyager CLI](../yb-voyager-cli/).
```
Comment (amit-yb) [inline docs/content/preview/migrate/install-yb-voyager.md:137]: We should drop this. The page already has a `Next step` directing users to `Migration steps`.

File: docs/content/preview/migrate/ubuntu.md [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+1. Install the Yugabyte apt repository on your machine using the following command:
+
+    ```sh
+    wget https://s3.us-west-2.amazonaws.com/downloads.yugabyte.com/repos/reporpms/yb-apt-repo_1.0.0_all.deb
+    sudo apt-get install ./yb-apt-repo_1.0.0_all.deb
+    ```
+
+    This repository contains the `yb-voyager` debian package and the dependencies required to run `yb-voyager`.
+
+2. Clean out the temporary cache of `apt-get` and update the package lists of `apt-get` on your machine using the following commands:
```
Comment (amit-yb) [inline docs/content/preview/migrate/ubuntu.md:18]: Drop the unnecessary words `out` and `temporary`.

File: docs/content/preview/migrate/ubuntu.md [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+    sudo apt-get clean
+    sudo apt-get update
+    ```
+
+3. Install yb-voyager and its dependencies using the following command:
+
+    ```sh
+    sudo apt-get install yb-voyager
+    ```
+
```
Comment (amit-yb) [inline docs/content/preview/migrate/ubuntu.md:30]: Check that yb-voyager is installed using the following command:

    yb-voyager version

File: docs/content/preview/migrate/install-yb-voyager.md [hunk 13]
```diff
@@ -20,10 +20,21 @@ The following sections describe the prerequisites for installing YugabyteDB Voya
...
 - MacOS (currently supported only for PostgreSQL source database)
 
+    {{< note title = "note" >}}
+
+You will get the following error if you use MySQL/Oracle as the source database on MacOS:
+
+```output
+"ora2pg" not found. Check if it is installed and included in the path.
+```
+
```
Comment (amit-yb) [inline docs/content/preview/migrate/install-yb-voyager.md:35]: Somewhere here suggest the user to use yb-voyager docker container to migrate from Oracle/MySQL on macOS.

File: docs/content/preview/migrate/docker.md [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+
+```sh
+docker volume create --driver local \
+      --opt type=none \
+      --opt device=[PATH_TO_YOUR_EXPORT_DIR] \
+      --opt o=bind \
+      export-dir
+```
+
+The volume created is named `export-dir`. You can use this name as the path by specifying `export-dir` without any path separators for your export directory while running yb-voyager.
```
Comment (amit-yb) [inline docs/content/preview/migrate/docker.md:40]: This entire section is not required anymore. The yb-voyager-docker script internally takes care of setting up volumes.

File: docs/content/preview/migrate/airgapped.md [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+
+    ```sh
+    // For RHEL 8 or CentOS 8
+    wget https://downloads.yugabyte.com/repos/airgapped/airgapped-rhel8.tar.gz
+    ```
+
+    ```sh
+    // For Ubuntu
+    wget https://downloads.yugabyte.com/repos/airgapped/airgapped_ubuntu.tar.gz
+    ```
```
Comment (amit-yb) [inline docs/content/preview/migrate/airgapped.md:26]: We should not give the installer bundles to the user. Instead, give them the list of steps to prepare the installer bundle. @ShivanshGahlot 

It is important that, the RHEL/CentOS machine used to prepare the installer bundle and the machine used to install voyager, MUST have exact same existing packages on them.

File: docs/content/preview/migrate/airgapped.md [hunk 16]
```diff
@@ -0,0 +1,46 @@
+<!--
```
Comment (amit-yb) [inline docs/content/preview/migrate/airgapped.md:1]: Also include the steps to install docker image in an airgapped environment (as shared in the dms-docs channel).

File: docs/content/preview/migrate/ubuntu.md (hunk 11 above)
Comment (polarweasel) [reply docs/content/preview/migrate/ubuntu.md:18]: Even better: "Clean the apt-get cache and package lists using the following commands:"

File: docs/content/preview/migrate/docker.md (hunk 14 above)
Comment (aishwarya24) [reply docs/content/preview/migrate/docker.md:40]: great , yes ill remove it.

File: docs/content/preview/migrate/install-yb-voyager.md (hunk 7 above)
Comment (aishwarya24) [reply docs/content/preview/migrate/install-yb-voyager.md:63]: Thats what i always thought Thanks for the clarification.

File: docs/content/preview/migrate/install-yb-voyager.md [hunk 17]
```diff
@@ -49,7 +60,7 @@ export EXPORT_DIR=$HOME/export-dir
 
 The export directory has the following sub-directories and files:
 
-- `reports` directory contains the generated *Schema Analysis Report*
+- `reports` directory contains the generated *Schema Analysis Report*.
```
[the remark this answers is not in the document]
Comment (aishwarya24) [reply docs/content/preview/migrate/install-yb-voyager.md:63]: Understood and I moved this section right after `create a user`: https://docs.yugabyte.com/preview/migrate/migrate-steps/#create-a-user

File: docs/content/preview/migrate/install-yb-voyager.md (hunk 13 above)
Comment (aishwarya24) [reply docs/content/preview/migrate/install-yb-voyager.md:35]: @amit , do we need to include the note. We can mention for MacOS as follows:

Comment (amit-yb) [reply docs/content/preview/migrate/install-yb-voyager.md:35]:
```
- macOS (For MySQL/Oracle source databases, [install yb-voyager](#install-yb-voyager) using the Docker option.)
```
should be enough.

File: docs/content/preview/migrate/migrate-steps.md [hunk 18]
```diff
@@ -108,6 +108,27 @@ DROP USER ybvoyager;
 
 {{< /warning >}}
 
+## Create an export directory
+
+yb-voyager keeps all of its migration state, including exported schema and data, in a local directory called the *export directory*.
+
+Before starting migration, you should create the export directory on a file system that has enough space to keep the entire data dump. Next, you should provide the path of the export directory as a mandatory argument (`--export-dir`) to each invocation of the yb-voyager command.
+
+yb-voyager uses the directory to store source data, schema files, and the migration state. The file system in which the directory resides must have enough free space to hold the entire source database. Create an export directory in the local file system on the machine where YugabyteDB Voyager will be installed, and place its path in an environment variable.
```
Comment (amit-yb) [inline docs/content/preview/migrate/migrate-steps.md:117]: This entire para reiterates what is already said in the previous two paragraphs. Consider removing it.

File: docs/content/preview/migrate/install-yb-voyager.md (hunk 13 above)
Comment (aishwarya24) [reply docs/content/preview/migrate/install-yb-voyager.md:35]: Thank you ! This is very clear.

**Pull request merged**